File: src/main/java/ru/ssau/tourism/entities/Tour.java
package ru.ssau.tourism.entities;

import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Setter
@NoArgsConstructor
@EqualsAndHashCode(of = {"id"})
public class Tour {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @Column
    private String title;
    @Column
    private Integer price;
    @Column
    private String info;

    public Long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public Integer getPrice() {
        return price;
    }

    public String getInfo() {
        return info;
    }
}

File: src/main/kotlin/ru/ssau/tourism/entities/Tour.kt
package ru.ssau.tourism.entities

import javax.persistence.*

@Entity
class Tour constructor(
		@Id
		@GeneratedValue(strategy = GenerationType.AUTO)
		var id: Long? = null,
		@Column
		var title: String? = null,
		@Column
		var price: Int? = null,
		@Column
		var info: String? = null
) {
	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other?.javaClass != javaClass) return false
		other as Tour
		if (id != other.id) return false
		return true
	}

	override fun hashCode(): Int {
		return id?.hashCode() ?: 0
	}
}
